### PR TITLE
chore(refactor): Move all custom API errors to errors.py

### LINKED ISSUE #7008: Remove duplicate exceptions in api/helpers/errors and exceptions.py
For some strange reason, duplicate errors like ForbiddenException, ConflictException and UnprocessableEntity are present in code in errors.py and exceptions.py. Remove duplicate types

### PR DESCRIPTION
**Pull request by Satyam52**
Fixes #7008

#### Short description of what this resolves:
removed duplicates in exception.py and the corresponding files which were importing exception from app/api/helpers/exception.py.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Clones removed
==============
+ app/api/tax.py  -1
+ app/api/tickets.py  -1
+ app/api/access_codes.py  -1
+ app/api/orders.py  -2
+ app/api/attendees.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821&bid=18150505)

File: app/api/helpers/ticketing.py [hunk 1]
```diff
@@ -6,7 +6,8 @@
 from flask_rest_jsonapi.exceptions import ObjectNotFound
 
 from app.api.helpers.db import get_count, save_to_db, safe_query_by_id
-from app.api.helpers.exceptions import ConflictException, UnprocessableEntity
+from app.api.helpers.exceptions import ConflictException
+from app.api.helpers.errors import UnprocessableEntityError
```
Comment (niranjan94) [inline app/api/helpers/ticketing.py:10]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/tax.py [hunk 2]
```diff
@@ -6,9 +6,12 @@
 from app.api.bootstrap import api
 from app.api.helpers.db import get_count, safe_query
 from app.api.helpers.exceptions import (
-    ConflictException,
-    ForbiddenException,
-    MethodNotAllowed,
+    ConflictException,
```
Comment (niranjan94) [inline app/api/tax.py:9]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/orders.py [hunk 3]
```diff
@@ -19,9 +19,10 @@
 )
 from app.api.helpers.errors import BadRequestError
 from app.api.helpers.exceptions import (
-    ConflictException,
-    ForbiddenException,
-    UnprocessableEntity,
+    ConflictException
+)
+from app.api.helpers.errors import (
```
Comment (niranjan94) [inline app/api/orders.py:24]: Codacy Issue found: [Unused ForbiddenError imported from app.api.helpers.errors](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/helpers/utilities.py [hunk 4]
```diff
@@ -9,7 +9,10 @@
 from flask import current_app
 from itsdangerous import Serializer
 
-from app.api.helpers.exceptions import UnprocessableEntity
+from app.api.helpers.errors import (
+    UnprocessableEntityError
+
```
Comment (niranjan94) [inline app/api/helpers/utilities.py:14]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/attendees.py [hunk 5]
```diff
@@ -8,8 +8,11 @@
 from app.api.helpers.db import safe_query
 from app.api.helpers.exceptions import (
     ConflictException,
-    ForbiddenException,
-    UnprocessableEntity,
+
```
Comment (niranjan94) [inline app/api/attendees.py:11]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/event_topics.py [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
 from app.api.bootstrap import api
 from app.api.helpers.db import safe_query
-from app.api.helpers.exceptions import UnprocessableEntity
+from app.api.helpers.errors import UnprocessableEntityError
```
Comment (niranjan94) [inline app/api/event_topics.py:7]: Codacy Issue found: [Unused UnprocessableEntityError imported from app.api.helpers.errors](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/access_codes.py [hunk 7]
```diff
@@ -6,9 +6,12 @@
 from app.api.helpers.db import safe_query
 from app.api.helpers.exceptions import (
     ConflictException,
```
Comment (iamareebjamal) [inline app/api/access_codes.py:8]: Why the love with ConflictException? Why leave it?

**Comment by iamareebjamal** [comment]
Syntactically wrong as well and the tests are also failing

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones removed
==============
+ app/api/attendees.py  -1
+ app/api/access_codes.py  -1
+ app/api/tickets.py  -1
+ app/api/tax.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821&bid=18150505)

File: app/api/event_topics.py (hunk 6 above)
Comment (niranjan94) [inline app/api/event_topics.py:7]: Codacy Issue found: [Unused UnprocessableEntityError imported from app.api.helpers.errors](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ app/api/tax.py  -1
+ app/api/tickets.py  -1
+ app/api/attendees.py  -1
+ app/api/access_codes.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821&bid=18150505)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ app/api/access_codes.py  -1
+ app/api/tickets.py  -1
+ app/api/tax.py  -1
+ app/api/attendees.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821&bid=18150505)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ app/api/tax.py  -1
+ app/api/attendees.py  -1
+ app/api/access_codes.py  -1
+ app/api/tickets.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821&bid=18150505)

File: app/api/access_codes.py [hunk 8]
```diff
@@ -1,13 +1,14 @@
...
 from flask_rest_jsonapi.exceptions import ObjectNotFound
 from sqlalchemy.orm.exc import NoResultFound
-
 from app.api.bootstrap import api
 from app.api.helpers.db import safe_query
 from app.api.helpers.exceptions import (
-    ConflictException,
-    ForbiddenException,
-    UnprocessableEntity,
+    ConflictException
```
Comment (iamareebjamal) [inline app/api/access_codes.py:7]: Why is this left?

Comment (Satyam52) [reply app/api/access_codes.py:7]: I didn't find conflictException in other file(error.py).

Comment (iamareebjamal) [reply app/api/access_codes.py:7]: Then move it there

Comment (Satyam52) [reply app/api/access_codes.py:7]: so I should remove all exceptions from exception.py and move to error.py.there are only two left ConflictException and MethodNotAllowed?

Comment (Satyam52) [reply app/api/access_codes.py:7]: @iamareebjamal Shall I move all exception from exception.py to error.py ???

Comment (iamareebjamal) [reply app/api/access_codes.py:7]: Yup

Comment (iamareebjamal) [reply app/api/access_codes.py:7]: Yup

**Comment by iamareebjamal** [comment]
Remove exceptions.py and resolve conflicts

**Comment by iamareebjamal** [comment]
There are still merge conflicts

**Comment by Satyam52** [comment]
@iamareebjamal resolved merge conflicts.

**Comment by iamareebjamal** [comment]
Great work but I don't think you have `pre-commit` set up correctly. I understand, unfortunately, there is no automated way to enable it automatically. So you should run `pre-commit install` so that it automatically reformats files when you commit things. 

Anyway I have created a PR to reformat files after your safe_query refactor https://github.com/fossasia/open-event-server/pull/7020, so once it is merged, please rebase your branch and resolve conflicts and run `black .` so that it automatically formats the files for you according to the standard. Then commit and push the changes.
Thanks

**Comment by Satyam52** [comment]
So, I should install pre-commit and then rebase with development branch and run blank . before commiting and pushing to my branch ?

**Comment by iamareebjamal** [comment]
Yup

**Comment by Satyam52** [comment]
Screenshot from 2020-05-20 19-23-07

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Clones added
============
- app/api/microlocations.py  1
- app/api/session_types.py  1
- app/api/faq_types.py  1
- app/api/tracks.py  1
- app/api/ticket_tags.py  1
         

Clones removed
==============
+ app/api/access_codes.py  -1
+ app/api/attendees.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821&bid=18150505)

File: app/api/session_types.py [hunk 9]
```diff
@@ -1,8 +1,9 @@
 from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
 
 from app.api.bootstrap import api
+from app.api.helpers.errors import ForbiddenError
 from app.api.helpers.db import safe_query_kwargs
-from app.api.helpers.exceptions import ForbiddenException
+from app.api.helpers.errors import ForbiddenError
```
Comment (niranjan94) [inline app/api/session_types.py:6]: Codacy Issue found: [Reimport 'ForbiddenError' (imported line 4)](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/faq_types.py [hunk 10]
```diff
@@ -1,8 +1,9 @@
 from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
 
 from app.api.bootstrap import api
+from app.api.helpers.errors import ForbiddenError
 from app.api.helpers.db import safe_query_kwargs
-from app.api.helpers.exceptions import ForbiddenException
+from app.api.helpers.errors import ForbiddenError
```
Comment (niranjan94) [inline app/api/faq_types.py:6]: Codacy Issue found: [Reimport 'ForbiddenError' (imported line 4)](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/feedbacks.py [hunk 11]
```diff
@@ -3,8 +3,9 @@
 from flask_rest_jsonapi.exceptions import ObjectNotFound
 
 from app.api.bootstrap import api
+from app.api.helpers.errors import ForbiddenError, UnprocessableEntityError
 from app.api.helpers.db import safe_query, safe_query_kwargs
-from app.api.helpers.exceptions import ForbiddenException, UnprocessableEntity
+from app.api.helpers.errors import ForbiddenError, UnprocessableEntityError
```
Comment (niranjan94) [inline app/api/feedbacks.py:8]: Codacy Issue found: [Reimport 'ForbiddenError' (imported line 6)](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/roles.py [hunk 12]
```diff
@@ -1,8 +1,9 @@
 from flask_rest_jsonapi import ResourceDetail, ResourceList
 
 from app.api.bootstrap import api
+from app.api.helpers.errors import UnprocessableEntityError
 from app.api.helpers.db import safe_query_kwargs
-from app.api.helpers.exceptions import UnprocessableEntity
+from app.api.helpers.errors import UnprocessableEntityError
```
Comment (niranjan94) [inline app/api/roles.py:6]: Codacy Issue found: [Reimport 'UnprocessableEntityError' (imported line 4)](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/microlocations.py [hunk 13]
```diff
@@ -1,8 +1,9 @@
 from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
 
 from app.api.bootstrap import api
+from app.api.helpers.errors import ForbiddenError
 from app.api.helpers.db import safe_query_kwargs
-from app.api.helpers.exceptions import ForbiddenException
+from app.api.helpers.errors import ForbiddenError
```
Comment (niranjan94) [inline app/api/microlocations.py:6]: Codacy Issue found: [Reimport 'ForbiddenError' (imported line 4)](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

File: app/api/attendees.py [hunk 14]
```diff
@@ -1,28 +1,27 @@
...
+from app.api.helpers.utilities import require_relationship
+from app.api.helpers.query import event_query
+from app.api.helpers.permissions import jwt_required
+from app.api.helpers.permission_manager import has_access
+from app.api.helpers.errors import (
+    ForbiddenError,
+    UnprocessableEntityError,
+    ConflictException,
+)
+from app.api.helpers.db import safe_query, safe_query_kwargs
```
Comment (iamareebjamal) [inline app/api/attendees.py:17]: Why this change? Revert

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones removed
==============
+ app/api/access_codes.py  -1
+ app/api/attendees.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821&bid=18150505)

File: app/api/roles.py (hunk 12 above)
Comment (niranjan94) [inline app/api/roles.py:6]: Codacy Issue found: [Reimport 'UnprocessableEntityError' (imported line 4)](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ app/api/access_codes.py  -1
+ app/api/attendees.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5536821&bid=18150505)

File: app/api/access_codes.py [hunk 15]
```diff
@@ -1,13 +1,13 @@
 from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
 from flask_rest_jsonapi.exceptions import ObjectNotFound
 from sqlalchemy.orm.exc import NoResultFound
-
```
Comment (iamareebjamal) [inline app/api/access_codes.py:4]: Why remove this line?

File: app/api/access_codes.py [hunk 16]
```diff
@@ -1,13 +1,13 @@
...
 from app.api.bootstrap import api
+
```
Comment (iamareebjamal) [inline app/api/access_codes.py:5]: Why add a line here?
